components/engine_tirocks/src/properties/mod.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+mod mvcc;
+mod range;
+mod table;
+mod ttl;
+
+use std::{
+    cmp,
+    collections::BTreeMap,
+    io::Read,
+    ops::{Deref, DerefMut},
+};
+
+use codec::{
+    number::NumberCodec,
+    prelude::{NumberDecoder, NumberEncoder},
+};
+use collections::HashMap;
+use tirocks::properties::table::user::UserCollectedProperties;
+
+pub use self::{
+    mvcc::MvccPropertiesCollectorFactory,
+    range::{RangeProperties, RangePropertiesCollectorFactory},
+    table::{RocksTablePropertiesCollection, RocksUserCollectedProperties},
+    ttl::TtlPropertiesCollectorFactory,
+};
+
+#[derive(Clone, Debug, Default)]
+pub struct IndexHandle {
```

Comment:
This struct is really hard to understand. From the code I don't see how it interacts with "stored block".

Reply:
The struct is copied from engine_rocks. It's named in the old days. Now I change it to `PropIndex`.